flaskbb/utils/helpers.py

```diff
@@ -111,87 +112,6 @@ def render_template(template, **context):  # pragma: no cover
     return render_theme_template(theme, template, **context)
 
 
-# TODO(anr): clean this up
-def do_topic_action(topics, user, action, reverse):  # noqa: C901
-    """Executes a specific action for topics. Returns a list with the modified
-    topic objects.
-
-    :param topics: A iterable with ``Topic`` objects.
-    :param user: The user object which wants to perform the action.
-    :param action: One of the following actions: locked, important and delete.
-    :param reverse: If the action should be done in a reversed way.
-                    For example, to unlock a topic, ``reverse`` should be
-                    set to ``True``.
-    """
-    if not topics:
-        return False
-
-    from flaskbb.utils.requirements import (
-        IsAtleastModeratorInForum,
-        CanDeleteTopic,
-        Has,
-    )
-
-    if not Permission(IsAtleastModeratorInForum(forum=topics[0].forum)):
-        flash(
-            _("You do not have the permissions to execute this action."),
-            "danger",
-        )
-        return False
-
-    modified_topics = 0
-    if action not in {"delete", "hide", "unhide"}:
-        for topic in topics:
-            if getattr(topic, action) and not reverse:
-                continue
-
-            setattr(topic, action, not reverse)
-            modified_topics += 1
-            topic.save()
-
-    elif action == "delete":
-        if not Permission(CanDeleteTopic):
-            flash(
-                _("You do not have the permissions to delete these topics."),
-                "danger",
-            )
-            return False
-
-        for topic in topics:
-            modified_topics += 1
-            topic.delete()
-
-    elif action == "hide":
-        if not Permission(Has("makehidden")):
-            flash(
-                _("You do not have the permissions to hide these topics."),
-                "danger",
-            )
-            return False
-
-        for topic in topics:
-            if topic.hidden:
-                continue
-            modified_topics += 1
-            topic.hide(user)
-
-    elif action == "unhide":
-        if not Permission(Has("makehidden")):
-            flash(
-                _("You do not have the permissions to unhide these topics."),
-                "danger",
-            )
-            return False
-
-        for topic in topics:
-            if not topic.hidden:
-                continue
-            modified_topics += 1
-            topic.unhide()
-
-    return modified_topics
```

Comment:
Nice refactoring. However, we need to check where we do the permissions check.
I could not find a place, where they have been moved.